src/Api/HL/RSQLInput.php

```diff
@@ -0,0 +1,303 @@
+                    $b = str_replace(array('%', '*'), array('.', '.*'), $b);
+                    return preg_match('/^' . $b . '$/i', $a);
+                },
+                'sql_where_callable' => function ($a, $b) use ($DB) {
+                    $b = str_replace(array('%', '*'), array('_', '%'), $b);
+                    return [
+                        [$DB::quoteName($a) => ['LIKE', $b]]
+                    ];
+                },
+            ],
+            [
+                'operator' => '=isnull=',
+                'description' => 'is null',
+                'callable' => fn ($a, $b) => $a === null,
+                'sql_where_callable' => fn ($a, $b) => [
+                    [$DB::quoteName($a) => null]
+                ],
+            ],
+            [
+                'operator' => '=notnull=',
+                'description' => 'is not null',
+                'callable' => fn ($a, $b) => $a !== null,
+                'sql_where_callable' => fn ($a, $b) => [
+                    ['NOT' => [$DB::quoteName($a) => null]]
+                ],
+            ],
+            [
+                // Empty string or null
+                'operator' => '=empty=',
+                'description' => 'is empty',
+                'callable' => fn ($a, $b) => empty($a),
+                'sql_where_callable' => fn ($a, $b) => [
+                    [
+                        'OR' => [
+                            [$DB::quoteName($a) => ''],
+                            [$DB::quoteName($a) => null],
+                        ]
+                    ]
+                ],
+            ],
+            [
+                'operator' => '=notempty=',
+                'description' => 'is not empty',
+                'callable' => fn ($a, $b) => !empty($a),
+                'sql_where_callable' => fn ($a, $b) => [
+                    [
+                        'AND' => [
+                            [$DB::quoteName($a) => ['<>', '']],
+                            [$DB::quoteName($a) => ['<>', null]],
+                        ]
+                    ]
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * @return array
+     * @phpstan-return {field: string, value: string, callable: callable}[]
+     */
+    private function getFilters()
+    {
+        $filters = [];
+        $rsql_filters = explode(';', $this->filter_string);
+        $operators = $this->getOperators();
+        $operator_pattern = '(' . implode('|', array_column($operators, 'operator')) . ')';
+        foreach ($rsql_filters as $rsql_filter) {
+            // filter pattern is field_name + operator + value where the operator always starts and ends with a symbol and the value may be a quoted string
+            [$field, $operator, $value] = preg_split("/$operator_pattern/", $rsql_filter, 2, PREG_SPLIT_DELIM_CAPTURE);
+            $value = trim($value, '\\?"\'');
```

Comment:
Can you explain why value is trimmed like this ?

Reply:
I don't even remember adding this. It doesn't seem useful though so I've removed it.